models/classes/xmlEditor/XmlEditor.php

```diff
@@ -59,6 +63,14 @@ public function saveStringTest(core_kernel_classes_Resource $test, string $testS
      */
     public function isLocked(): bool
     {
+        $userSettings = $this->getUserSettingsService()->getCurrentUserSettings();
+        if (
+            $this->getFeatureFlagChecker()->isEnabled(FeatureFlagCheckerInterface::FEATURE_FLAG_SOLAR_DESIGN_ENABLED)
+            && $userSettings->getSetting(UserSettingsInterface::INTERFACE_MODE) == GenerisRdf::PROPERTY_USER_INTERFACE_MODE_SIMPLE
+        ) {
+            return true;
+        }
```

Comment:
Do we really need to check the feature flag in all those places? It is not really necessary, you will only build this settings in `UserSettingsService` if the feature flag is enabled, there is the actual place you should care about the feature flag. 

```suggestion
        if ($this->getUserSettingsService()->getCurrentUserSettings()->getSetting(UserSettingsInterface::INTERFACE_MODE) === GenerisRdf::PROPERTY_USER_INTERFACE_MODE_SIMPLE) {
            return true;
        }
```

Reply:
Do you mean that we shouldn't return the actual value of the interface mode from getSetting if the feature flag is not set ?

Reply:
I mean, this setting will be returning `null` if the feature flag does not exist. So we populate those settings based on the feature flag instead of checking the feature flag all the time combined with the feature flag in every place